cluster/photon-controller/util.sh

```diff
@@ -0,0 +1,896 @@
+        kube::log::error "Please start ssh-agent, add your identity, and retry."
+        exit 1
+    fi
+}
+
+#
+# Verify that Photon Controller has been configured in the way we expect. Specifically
+# - Have the flavors been created?
+# - Has the image been uploaded?
+# TODO: Check the tenant and project as well.
+function verify-photon-config {
+    kube::log::status "Validating Photon configuration..."
+
+    fail=0
+    rc=0
+    $PHOTON flavor list | awk -Ft '{print $2}' | grep "^${PHOTON_MASTER_FLAVOR}$" > /dev/null 2>&1 || rc=$?
+    if [ $rc -ne 0 ]; then
+        kube::log::error "Cannot find flavor named ${PHOTON_MASTER_FLAVOR}"
+        fail=1
+    fi
+
+    if [ "${PHOTON_MASTER_FLAVOR}" != "${PHOTON_NODE_FLAVOR}" ]; then
+        rc=0
+        $PHOTON flavor list | awk -Ft '{print $2}' | grep "^${PHOTON_NODE_FLAVOR}$" > /dev/null 2>&1 || rc=$?
+        if [ $rc -ne 0 ]; then
+            kube::log::error "Cannot find flavor named ${PHOTON_NODE_FLAVOR}"
+            fail=1
+        fi
+    fi
+
+    rc=0
+    $PHOTON image list | awk -Ft '{print $1}' | grep "^${PHOTON_IMAGE_ID}$"  > /dev/null 2>&1 || rc=$?
+    if [ $rc -ne 0 ]; then
+        kube::log::error "Cannot find image named ${PHOTON_IMAGE_ID}"
+        fail=1
+    fi
+
+    if [ $fail -eq 1 ]; then
+        exit 1
+    fi
+}
+
+#
+# Verifies that a given command is in the PATH
+#
+function verify-cmd-in-path {
+    cmd=$1
+    which ${cmd} >/dev/null || {
+        kube::log::error "Can't find ${cmd} in PATH, please install and retry."
+        exit 1
+    }
+}
+
+#
+# Checks that KUBE_TEMP is set, or sets it
+# If it sets it, it also creates the temporary directory
+# and sets up a trap so that we delete it when we exit
+#
+function ensure-temp-dir {
+  if [[ -z ${KUBE_TEMP-} ]]; then
+    KUBE_TEMP=$(mktemp -d -t kubernetes.XXXXXX)
+    trap-add 'rm -rf "${KUBE_TEMP}"' EXIT
+  fi
+}
+
+#
+# Waits for a remote process to be running
+# We use this to wait for salt to start
+#
+function wait-for-remote-process {
+    local vm_name=$1
+    local vm_ip=$2
+    local process_name=$3
+    local max_attempt=600
+
+    kube::log::status "Waiting up to 10 minutes for ${process_name} to start on ${vm_name}..."
+
+    local attempt=0
+    while true; do
+        local rc=0
+        run-ssh-cmd -i ${vm_ip} "pgrep ${process_name}" || rc=1
+        if [[ $rc != 0 ]]; then
+            if (( attempt == max_attempt )); then
+                kube::log::error "${process_name} on ${vm_ip} did not start after ten minutes"
+                kube::log::error "Cannot proceed: you may need to retry to log into the VM to debug"
+                exit 1
+            fi
+        else
+            break
+        fi
+        attempt=$((attempt+1))
+        sleep 1
+    done
+}
+
+#
+# Repeatedly try a command over ssh until it succeeds or until five minutes have passed
+# The timeout isn't exact, since we assume the command runs instantaneously, and
+# it doesn't.
+#
+function try-until-success-ssh {
```

Comment:
Only difference in the following two functions is that of VM IP. Can you merge them somehow instead of repeating the code?